docs/readme-developer.md

```diff
@@ -1,111 +1,184 @@
+
+## Building the game
+
+### Building from the command line
 
-To compile and also run unit tests, the "test" target can be used:
+(Note: This commands require CMake 3.21+. If you are using a lower version of CMake you will not be able to use the presets mentioned in this section.)
 
+Here's a summary of every command you will need for development:
+
+```bash
+$ cmake --preset <preset>                                       # configure project (only needs to be done once)
+$ cmake --build --preset <preset>-debug                         # build Endless Sky and all tests
+$ cmake --build --preset <preset>-debug --target EndlessSky     # build only the game
+$ ctest --preset <preset>-test                                  # run the unit tests
+$ ctest --preset <preset>-benchmark                             # run the benchmarks
+$ ctest --preset <preset>-integration                           # run the integration tests (Linux only)
 ```
-  $ scons test
+
+The executable will be located in `build/<preset>/Debug/`. If you'd like to debug a specific integration test (on any OS), you can do so as follows:
+
+```bash
+$ ctest --preset <preset>-integration-debug -R <name>
 ```
 
-The produced test binary will be invoked with a reasonable set of arguments. To run any test benchmarks, you will need to invoke the test binary directly. Refer to the build pipeline definitions for the current required arguments.
-More information on unit tests can be found in the project's "tests" directory, or by running the binary with the `--tests` flag.
+You can get a list of integration tests with `ctest --preset <preset>-integration-debug -N`.
+
+(You can also use the `<preset>-release` preset for a release build, and the output will be in the Release folder).
 
+Replace `<preset>` with one of the following presets:
 
+- Windows: `clang-cl` (builds with Clang for Windows), `mingw` (builds with MinGW), `mingw32` (builds with x86 MinGW)
+- MacOS: `macos` or `macos-arm` (builds with the default compiler, for x64 and ARM64 respectively)
+- Linux: `linux` (builds with the default compiler), `linux-gles` (compiles with GLES instead of OpenGL support)
 
-## Windows
+You can list all of available presets with `cmake --list-presets`.
 
-MinGW provides the tools used to compile the source code into the game's executable. For most cases, you should download the [MinGW-W64 Online Installer](https://sourceforge.net/projects/mingw-w64/files/Toolchains%20targetting%20Win32/Personal%20Builds/mingw-builds/installer/mingw-w64-install.exe). When installing, use the following settings:
+### Using an IDE
 
-- Version: `8.1.0` (or greater; this document assumes 8.1.0)
-- Architecture: `x86_64`
-- Threads: `posix`
-- Exception: `seh`
+Most IDEs have CMake support, and can be used to build the game. We recommend using [Visual Studio Code](#visual-studio-code).
 
-Endless Sky requires precompiled libraries to compile and play: [Download link](https://endless-sky.github.io/win64-dev.zip)
```

Comment:
I'm not sure if this is needed, but it's also not in the cmake guide. I think this is required without vcpkg?

Reply:
Isn't this like outdated a few years now? Also if something doesn't meet minimal requirements of vcpkg, I wouldn't expect CMake to work there either.

Reply:
~~ah yes typical microsoft quality~~

Reply:
I (re)added this line.